consensus/src/event_processor.rs

```diff
@@ -634,6 +633,15 @@ impl<T: Payload> EventProcessor<T> {
             .get_valid_proposers(proposal_round + 1);
         debug!("{}Voted: {} {}", Fg(Green), Fg(Reset), vote);
 
+        self.last_vote_sent.replace(vote.clone());
+        if let Err(e) = self
+            .storage
+            .save_vote(&vote)
+            .context("Fail to persist last vote")
+        {
+            error!("{:?}", e);
+            return;
+        }
```

Comment:
I am not sure moving this accomplishes anything and adds more lines of code.

Reply:
I was trying to make it consistent - store the vote before sent to network. before this PR we don't persist timeout vote so it sends to safety rules over and over again